Makefile

```diff
@@ -28,4 +29,36 @@ tidy:
 
 clean:
 	$(GO) clean -i ./...
-	rm -f vt
\ No newline at end of file
+	rm -f vt
+
+# Pretty: formats the code using gofumpt and goimports-reviser
+pretty: install-tools
+	@echo "Running formatting tools..."
+	@gofumpt -l -w . >/dev/null 2>&1 || true
+	@goimports-reviser -project-name $$(go list -m) -rm-unused -set-alias -format . >/dev/null 2>&1 || true
+
+# Install tools: Checks if the required tools are installed, installs if missing
+install-tools:
+	@command -v gofumpt >/dev/null 2>&1 || { \
+		echo "Installing gofumpt..."; \
+		go install mvdan.cc/gofumpt@latest >/dev/null 2>&1; \
+	}
+	@command -v goimports-reviser >/dev/null 2>&1 || { \
+		echo "Installing goimports-reviser..."; \
+		go install github.com/incu6us/goimports-reviser@latest >/dev/null 2>&1; \
+	}
+	@command -v golangci-lint >/dev/null 2>&1 || { \
+		echo "Installing golangci-lint..."; \
+		curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH)/bin $(GOLANGCI_LINT_VERSION); \
+	}
+
+# Lint: runs golangci-lint
+lint: install-tools
+	@echo "Running golangci-lint..."
+	@golangci-lint run --config .golangci.yml ./go/...
+
+install-hooks:
+	@echo "Installing Git hooks..."
+	@cp git-hooks/pre-commit .git/hooks/pre-commit
```

Comment:
I think that this might conflict with Vitess where we make it a symlink. That is done in `build.env` which is sourced in `dev.env` which is sourced in `bootstrap.sh`, which is run when you `make tools` which is a makefile target called by many others (as it installs the deps). We could use a symlink here as well, no? Then the symlink would get changed when switching between the two repos.

```
build.env:ln -sf "$PWD/misc/git/pre-commit" .git/hooks/pre-commit
```